server/sublist.go

```diff
@@ -621,6 +615,49 @@ func (s *Sublist) match(subject string, doLock bool, result *SublistResult) *Sub
 	return result
 }
 
+func (s *Sublist) hasInterest(subject string, doLock bool) bool {
+	// Check cache first.
+	if doLock {
+		s.RLock()
+	}
+	var matched bool
+	if s.cache != nil {
+		if r, ok := s.cache[subject]; ok {
+			matched = len(r.psubs)+len(r.qsubs) > 0
```

Comment:
If there was no distinction of psubs/qsubs, the mere presence of an entry is enough, that is, the `ok` already means that there is interest. But I am of the opinion that we would need to check for the user check intention.

Reply:
Since @derekcollison seem to say that we don't need to have a specialized HasInterest() call, then I would have replaced this all "if" statement with:
```
if s.cache != nil {
    _, matched = s.cache[subject]
}
```

Reply:
Makes sense, done.